### ISSUE TITLE
[BUG] Improve type compatibility with newer ESLint versions

### ISSUE THREAD

**Issue by fasttime** [yeonjuan/html-eslint#448]
**Describe the bug**

`@html-eslint/eslint-plugin` includes type definitions that aren't compatible with newer versions of ESLint.
This is because it uses `@eslint/plugin-kit` v0.3.1 - v0.3.5, which relies on older versions of `@eslint/core` types that ESLint no longer supports.

This issue was originally reported at eslint/eslint#20287.

**To Reproduce**

This config file demonstrates the incompatibility with the `ESLint.Plugin` type:

```ts
import html from '@html-eslint/eslint-plugin';
import { defineConfig } from 'eslint/config';

export default defineConfig({
  files: ['**/*.html'],
  plugins: { html },
  language: 'html/html',
  extends: ['html/recommended'],
});
```

Here is a repro based on the original issue: https://stackblitz.com/edit/stackblitz-starters-iitfrc9w.

**Expected behavior**

I imagine the types should be compatible with ESLint v9.39.1 and possibly previous versions (currently, the lowest suppoerted version should be v9.27.0).

https://github.com/yeonjuan/html-eslint/blob/0b1787b8c95f0cd85b0531e90c7b9fae922fccd2/packages/eslint-plugin/package.json#L55

**Additional context**

ESLint v10.0.0 will adopt `@eslint/core` v1.x types. This ensures that future updates to core types will be reflected in previous versions of ESLint, since new minor releases (e.g., after implementing a new feature) will be picked up by the dependency range `"^1.x.y"` in `@eslint/plugin-kit` and in plugins that use a similar caret range. This should mitigate issues caused by different versions of the `@eslint/core` dependency.

**Comment by yeonjuan** [comment]
Hi @fasttime ,  Thank you for the report. Let me fix it!

**Comment by yeonjuan** [comment]
@fasttime It's fixed in 0.49.0. Thanks! https://stackblitz.com/edit/stackblitz-starters-ccneev6o

**Comment by fasttime** [comment]
Thanks! I checked and the types are fine now.
